kinto/core/utils.py

```diff
@@ -445,6 +445,11 @@ def view_lookup(request, uri):
     :rtype: tuple
     :returns: the resource name and the associated matchdict.
     """
+
+    # If we are accessing the top level listing of /buckets, there is no id
+    if (uri == '/buckets'):
+        return '', {id: False}
```

Comment:
I'm pretty sure `view_lookup(request, '/buckets')` should already work as expected, ie. return `"bucket", {}`

If it does not, then we can fix it. You could add a unit test in test_utils to assert that BTW

Reply:
Yes `view_lookup` does return `"bucket", {}` without that code, the issue I'm seeing is that I need it to return `'', {}` for this:

https://github.com/Kinto/kinto/blob/6d77dcb2b94f9198fa08db489776440f7df0c275/kinto/views/permissions.py#L123

`perms_descending_tree` is populated with:

https://github.com/Kinto/kinto/blob/6d77dcb2b94f9198fa08db489776440f7df0c275/kinto/authorization.py#L26

So I need `''` for `resource_name` in the `obtained = perms_descending_tree[resource_name][perm]` code above.

That's what's causing some of the back and forth between assigning `bucket` and `''` in some of my other changes.

Reply:
Ok, I get it.

So we should keep the `view_lookup()` consistent with what it does. `/buckets` is the `bucket` resource, so it should return.

Now, what we want is that to read `perms_descending_tree[""]["bucket:create"]`. So we should manage this corner here, and not in `view_lookup()`.

For example:
```py
if resource == "bucket" and permission == "create":
    obtained = perms_descending_tree[""]["bucket:create"]
```

That way we don't have to change `kinto.authorization.PERMISSIONS_INHERITANCE_TREE`, and we dont' change the semantics of `view_lookup()` for that particular. Just imagine that `view_lookup()` can be used by any other component and would not expect this `"", {"id": False}` response